tensorboard/webapp/widgets/linked_time_fob/types.ts

```diff
@@ -0,0 +1,46 @@
+/* Copyright 2022 The TensorFlow Authors. All Rights Reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+==============================================================================*/
+
+import {PluginType} from '../../metrics/types';
+import {ScaleLinear, ScaleTime} from '../../third_party/d3';
+import {Scale} from '../line_chart_v2/lib/public_types';
+
+export {Scale} from '../line_chart_v2/lib/public_types';
+
+export enum AxisDirection {
+  HORIZONTAL,
+  VERTICAL,
+}
+
+export enum Fob {
+  NONE,
+  START,
+  END,
+}
+
+export type TemporalScale =
+  | ScaleLinear<number, number>
+  | ScaleTime<number, number>;
+
+export interface FobCardData {
```

Comment:
Now that I am wondering do we want to allow both keys existing in `FobCardData`?  Probably not.
Sorry I did not think about this clearer. Now what I think is better is
```
interface HistogramCardFobData {
    scale: TemporalScale;
    steps: number[];
}

interface ScalarCardFobData {
    scale: Scale;
    minMax: [number, number];
}

type FobCardData =  HistogramCardFobData | ScalarCardFobData
```

so then instead of checking this.fobCardData.histogram,
we could have an attribute serves as the fobCard type
some pseudo code could be like 
```
private pluginType: PluginType = PluginType.NONE;

if (typeof(fobCardData) === HistogramCardFobData) {
pluginType =  PluginType.HISTOGRAM
}
if (typeof(fobCardData) === ScalarCardFobData) {
pluginType =  PluginType.SCALAR
}
```

then instead of checking `this.fobCardData.historgram` we check `pluginType === PluginType.HISTOGRAM`

Reply:
so after rethink this, I think define the axis is better than define the plugin. Using `if(Axis.VERTICAL)` is more meaningful then `if(.. HISTOGRAM)`. Please ignore the second part of the comment above. Instead,
```
private axis: AxisDirection;

switch(typeof(fobCardData)) {
    case HistogramCardFobData:
         axis = AxisDirection.VERTICAL;
    case ScalarCardFobData:
         axis = AxisDirection.HORIZONTAL
}
```
You can put it in `ngOnInit`